improver_tests/convection/test_ConvectionRatioFromComponents.py

```diff
@@ -0,0 +1,184 @@
+    """Global grid template"""
+
+    data = np.zeros((19, 37), dtype=np.float32)
+    cubes = CubeList([])
+    cubes.append(
+        set_up_variable_cube(
+            data.copy(),
+            name="convective_precipitation_rate",
+            units="m s-1",
+            grid_spacing=10,
+            domain_corner=(-90, -180),
+            attributes=GLOBAL_ATTRIBUTES,
+        )
+    )
+    cubes.append(
+        set_up_variable_cube(
+            data.copy(),
+            name="dynamic_precipitation_rate",
+            units="m s-1",
+            grid_spacing=10,
+            domain_corner=(-90, -180),
+            attributes=GLOBAL_ATTRIBUTES,
+        )
+    )
+    return cubes
+
+
+@pytest.fixture(name="uk_grid")
+def uk_grid_fixture() -> Cube:
+    """UK grid template"""
+
+    data = np.zeros((21, 22), dtype=np.float32)
+    cubes = CubeList([])
+    cubes.append(
+        set_up_variable_cube(
+            data.copy(),
+            name="convective_precipitation_rate",
+            units="m s-1",
+            spatial_grid="equalarea",
+            grid_spacing=96900.0,
+            domain_corner=(-1036000.0, -1158000.0),
+            attributes=UK_ATTRIBUTES,
+        )
+    )
+    cubes.append(
+        set_up_variable_cube(
+            data.copy(),
+            name="dynamic_precipitation_rate",
+            units="m s-1",
+            spatial_grid="equalarea",
+            grid_spacing=96900.0,
+            domain_corner=(-1036000.0, -1158000.0),
+            attributes=UK_ATTRIBUTES,
+        )
+    )
+    return cubes
+
+
+@pytest.mark.parametrize("grid_fixture", ["global_grid", "uk_grid"])
+def test_basic(request, grid_fixture):
+    """Ensure Plugin returns object of correct type and meta-data and that
+    no precip => masked array"""
+    grid = request.getfixturevalue(grid_fixture)
+    expected_array = np.ma.masked_all_like(grid[0].data)
+    result = ConvectionRatioFromComponents()(grid.copy())
+    assert isinstance(result, iris.cube.Cube)
+    assert_allclose(result.data, expected_array)
+    assert result.attributes == grid[0].attributes
+    assert result.long_name == "convective_ratio"
+    assert result.units == "1"
+
+
+# These tuples represent one data point. The first two values are the convective and
+# dynamic precipitation rate respectively. The last value is the expected result.
+@pytest.mark.parametrize(
+    "data_con_dyn_out",
+    [
+        (1.0, 0.0, 1.0),
+        (1.0, 1, 0.5),
+        (0.0, 1.0, 0.0),
+        (0.9e-9, 0.0, np.inf),
+        (1.1e-9, 0.0, 1.0),
+        (0.9e-9, 0.9e-9, 0.5),
+    ],
+)
+@pytest.mark.parametrize("grid_fixture", ["global_grid", "uk_grid"])
+@pytest.mark.parametrize("units", ["m s-1", "mm h-1"])
+def test_data(request, grid_fixture, data_con_dyn_out, units):
+    """Test that the data are calculated as expected for a selection of values on both
+    grids with SI and non-SI units including either side of the minimum precipitation
+    rate tolerance 1e-9 m s-1"""
```

Comment:
It might be useful to mention that you're modifying a single grid point to have non-zero values, as appropriate for testing, for the convective and dynamic precipitation ratios and all other grid points will be masked within the result array. That might make it more obvious to know what is being tested.

Reply:
Good point. I've extended this doc-string.